arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -508,6 +440,168 @@ public static VectorizedReader<?> replaceWithMetadataReader(
     return reader;
   }
 
+  private static final class LogicalTypeVisitorResult {
```

Comment:
```suggestion
  private static class LogicalTypeVisitorResult {
```

Reply:
ah nvm, I thought we're avoiding using `final` on classes in the codebase but it appears that we have plenty of examples of final classes